packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -104,18 +105,26 @@ export function maybeStripNodeJsVirtualPrefix(
 }
 
 /**
- * Convert any virtual module Id that was generated by the aliases returned from `getNodeCompatAliases()`
- * back to real a module Id and whether it is an external (built-in) package or not.
+ * Resolve the source import to an absolute path, potentially via its alias.
  */
-export function dealiasVirtualNodeJSImport(source: string) {
+export function resolveNodeJSImport(source: string) {
 	const alias = env.alias[source];
-	assert(
-		alias,
-		`Expected "${source}" to have a Node.js compat alias, but none was found`
-	);
-	assert(
-		!env.external.includes(alias),
-		`Unexpected unenv alias to external module: ${source} -> ${alias}`
-	);
-	return alias;
+	if (alias) {
+		assert(
+			!env.external.includes(alias),
+			`Unexpected unenv alias to external module: ${source} -> ${alias}`
+		);
+		source = alias;
+	}
+	// Resolve to an absolute path using the path to this file as the resolution starting point.
+	// This is essential so that we can find the `@cloudflare/unenv-preset` and `unenv` packages,
+	// which are dependencies of this package but may not be direct dependencies of the user's project.
+	const resolved = resolvePathSync(source, {
+		url: import.meta.url,
```

Comment:
Why not using the [unenv resolution](https://github.com/unjs/unenv/blob/008bf1c19a716faa5f2cf86d58d8531447d0e230/lib/index.d.mts#L51)?

Reply:
Yeah good question.

I realise the design of the unenv resolution stuff is supposed to aid combining different presets in different packages with the unenv library itself but in practice I have found that it adds extra complexity to the code and is hidden logic that is not always easy to reason about.

While trying to get this stuff working I switched to controlling the resolution directly because it was easier to be sure how it was working.

As far as I can tell doesn't appear to be needed for our purposes. It maybe that there are other use cases where it is valuable but not here.